filter/unquote_resolvers.py

```diff
@@ -0,0 +1,79 @@
+from typing import Any, Dict, List, Union
+
+import yaml
+from yaml.nodes import ScalarNode
+
+try:
+    from yaml import CSafeDumper as SafeDumper
+except ImportError:
+    from yaml import SafeDumper  # type: ignore
+
+
+class CustomDumper(SafeDumper):
+    pass
+
+
+def unquoted_tag_representer(dumper: CustomDumper, data: "Tag") -> ScalarNode:
+    return dumper.represent_scalar(data.tag, data.value, style="")
+
+
+class Tag:
+    def __init__(self, tag: str, value: str) -> None:
+        self.tag = tag
+        self.value = value
+
+
+CustomDumper.add_representer(Tag, unquoted_tag_representer)
+
+
+def unquote_resolvers(
+    data: Union[Dict, List],
+    indent: int = 2,
+    output_indent: int = 0,
+    *args: Any,
+    **kw: Any
+) -> str:
+    """
+    Remove quotes around resolver expressions and
+    produce nice YAML.
+    """
+
+    def process_item(item: Any) -> Any:
+        if isinstance(item, str) and item.startswith("!"):
+            tag_name, tag_body = item.split(" ", 1)
+            return Tag(tag_name, tag_body)
+        elif isinstance(item, dict):
+            return process_dict(item)
+        elif isinstance(item, list):
+            return process_list(item)
+        else:
+            return item
+
+    def process_dict(dct: Dict) -> Dict:
+        return {key: process_item(value) for key, value in dct.items()}
+
+    def process_list(lst: List) -> List:
+        return [process_item(value) for value in lst]
```

Comment:
For cases like this, I like using `functools.singledispatch`. It's the exact usecase.

```suggestion
    @singledispatch
    def process_item(item: Any) -> Any:
        """This is your fallback for non-registered types"""
        return item 
    
    @process_item.register
    def process_str(item: str):
            if item.startswith("!"):
                tag_name, tag_body = item.split(" ", 1)
                return Tag(tag_name, tag_body)
            return item
    
    @process_item.register
    def process_dict(item: dict) -> dict:
        return {key: process_item(value) for key, value in item.items()}
    
    @process_item.register
    def process_list(item: list) -> list:
        return [process_item(value) for value in item]
```

Reply:
ok done